### ISSUE TITLE
Custom Authentication Methods

### ISSUE THREAD

**Issue by ikogan** [sjarvela/kloudspeaker#164]
I'm trying to build a plugin to auto create users who login with the "remote" authentication method. The only way I've found to do that without hacking on delivered code is to do it during the `setup()` phase of plugin initialization. This means I have to check whether or not users exist and their validate their configuration every time a page loads and cannot utilize the session.

Would it be desirable to be able to pull authentication methods out into a plugin so that it can be customized a little better. Additionally, perhaps abstracting the way user/group information is loaded so that it can be replaced more easily would be beneficial as well.

**Comment by sjarvela** [comment]
You are right, backend modularisation (and implementation in general) is not good enough. I've been planning entire backend rewrite with some modern PHP framework, this would help getting basic things without doing everything in app.

I'll see if I can change the way how remote auth is handled, so that it can be customized.

**Comment by ikogan** [comment]
If you're planning on rewriting it, have you considered moving to NodeJS? There are some pretty nice frameworks out there for this.

**Comment by sjarvela** [comment]
Sure NodeJS would have a whole lot of different possibilities, but PHP has been the choice based on the fact that most service providers support PHP, unfortunately.

Choosing something like NodeJS, python or ruby would narrow down the potential users quite heavily.

**Comment by ikogan** [comment]
That's true, I'm not sure what Kloudspeaker's install base is like but I imagine it would open up some possibilities as well.
